hook-producer/src/handlers/mod.rs

```diff
@@ -1,15 +1,164 @@
 use axum::{routing, Router};
+use metrics_exporter_prometheus::PrometheusHandle;
 
-mod index;
+use hook_common::pgqueue::PgQueue;
 
-pub fn router() -> Router {
-    let recorder_handle = crate::metrics::setup_metrics_recorder();
+mod index;
+mod webhook;
 
+pub fn app(pg_pool: PgQueue, metrics: Option<PrometheusHandle>) -> Router {
     Router::new()
         .route("/", routing::get(index::get))
         .route(
             "/metrics",
-            routing::get(move || std::future::ready(recorder_handle.render())),
+            routing::get(move || match metrics {
+                Some(ref recorder_handle) => std::future::ready(recorder_handle.render()),
+                None => std::future::ready("no metrics recorder installed".to_owned()),
+            }),
```

Comment:
Metrics can only be installed once, so the pre-existing code fucked up tests (more than 1 test = panic). I made it optional so tests don't have to set it up at all.

Reply:
Thanks, I checked that out but `if cfg!(test)` seems like it would introduce the same `Some` vs `None` code that I need to have here.

I guess it would let me optionally construct the metrics thing inside the `app()` function, but I kind of like to create these dependencies in `main` anyway. Happy to change it if you think I should.

Reply:
We can roll it out like this and if it becomes to annoying revisit it in the future :+1: